src/wire.rs

```diff
@@ -35,43 +35,55 @@ impl Read for Aggregation {
 
     fn read_cfg(buf: &mut impl Buf, _: &()) -> Result<Self, Error> {
         let round = u64::read(buf)?;
-        
+
         let var1_len = u32::read(buf)? as usize;
         if buf.remaining() < var1_len {
             return Err(Error::EndOfBuffer);
         }
         let mut var1_bytes = vec![0u8; var1_len];
         buf.copy_to_slice(&mut var1_bytes);
-        let var1 = String::from_utf8(var1_bytes).map_err(|_| Error::Invalid("var1", "decoding from utf8 failed"))?;
+        let var1 = String::from_utf8(var1_bytes)
+            .map_err(|_| Error::Invalid("var1", "decoding from utf8 failed"))?;
 
         let var2_len = u32::read(buf)? as usize;
         if buf.remaining() < var2_len {
             return Err(Error::EndOfBuffer);
         }
         let mut var2_bytes = vec![0u8; var2_len];
         buf.copy_to_slice(&mut var2_bytes);
-        let var2 = String::from_utf8(var2_bytes).map_err(|_| Error::Invalid("var2", "decoding from utf8 failed"))?;
+        let var2 = String::from_utf8(var2_bytes)
+            .map_err(|_| Error::Invalid("var2", "decoding from utf8 failed"))?;
 
         let var3_len = u32::read(buf)? as usize;
         if buf.remaining() < var3_len {
             return Err(Error::EndOfBuffer);
         }
         let mut var3_bytes = vec![0u8; var3_len];
         buf.copy_to_slice(&mut var3_bytes);
-        let var3 = String::from_utf8(var3_bytes).map_err(|_| Error::Invalid("var3", "decoding from utf8 failed"))?;
-        
+        let var3 = String::from_utf8(var3_bytes)
+            .map_err(|_| Error::Invalid("var3", "decoding from utf8 failed"))?;
+
```

Comment:
this whole chunk could be optimized by using native read impls similar to #3, i could take that up